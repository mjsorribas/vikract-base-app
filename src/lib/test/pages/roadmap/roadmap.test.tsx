import { ChakraProvider } from "@chakra-ui/react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { expect, test, vi } from "vitest";

// Mock de una dependencia específica
vi.mock("@chakra-ui/react", async () => {
  const actual = await vi.importActual("@chakra-ui/react");
  return {
    ...actual,
    useBreakpointValue: () => "mockedValue",
    extendTheme: actual.extendTheme, // Asegúrate de incluir extendTheme en el mock
  };
});

import Roadmap from "lib/pages/roadmap/index";

const toJson = (component: renderer.ReactTestRenderer) => {
  const result = component.toJSON();
  expect(result).toBeDefined();
  return result as renderer.ReactTestRendererJSON;
};

test("Roadmap", () => {
  const component = renderer.create(
    <MemoryRouter initialEntries={[{ pathname: "/" }]}>
      <ChakraProvider>
        <Roadmap />
      </ChakraProvider>
    </MemoryRouter>
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
