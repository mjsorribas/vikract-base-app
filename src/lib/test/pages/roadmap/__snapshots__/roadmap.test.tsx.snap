// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Roadmap 1`] = `
[
  <div
    className="css-49pdgh"
  >
    <h2
      className="chakra-heading css-1dklj6k"
    >
      Our Roadmap
    </h2>
    <p
      className="chakra-text css-0"
    >
      <strong>
        2023:
      </strong>
      <ul
        className="css-155za0w"
        role="list"
      >
        <li
          className="css-0"
        >
          Create base app template
        </li>
        <li
          className="css-0"
        >
          Design flow to deploy to differents providers
        </li>
        <li
          className="css-0"
        >
          Design example pages
        </li>
        <li
          className="css-0"
        >
          Develop custom components
        </li>
      </ul>
    </p>
    <p
      className="chakra-text css-0"
    >
      <strong>
        2024:
      </strong>
      <ul
        className="css-155za0w"
        role="list"
      >
        <li
          className="css-0"
        >
          Create library with custom components
        </li>
        <li
          className="css-0"
        >
          Refactor custom components and move to library
        </li>
        <li
          className="css-0"
        >
          Start a new way to generate templates based on color tokens
        </li>
      </ul>
    </p>
  </div>,
  <span />,
  <span
    hidden={true}
    id="__chakra_env"
  />,
]
`;
